analyzers/src/SonarAnalyzer.CSharp/Rules/ArrayPassedAsParams.cs

```diff
@@ -40,7 +40,12 @@ public sealed class ArrayPassedAsParams : ArrayPassedAsParamsBase<SyntaxKind, Ar
             : null;
 
     protected override ITypeSymbol ArrayElementType(ArgumentSyntax argument, SemanticModel model) =>
-        model.GetTypeInfo(((ArrayCreationExpressionSyntax)argument.Expression).Type.ElementType).Type;
+        argument switch
+        {
+            _ when argument.Expression as ArrayCreationExpressionSyntax is { } arrayCreation => model.GetTypeInfo(arrayCreation.Type.ElementType).Type,
+            _ when argument.Expression as ImplicitArrayCreationExpressionSyntax is { } implicitArrayCreation => model.GetTypeInfo(implicitArrayCreation).Type,
```

Comment:
This looks wrong. It does return the array type, but the method is called ArrayElementType. It should be `model.GetTypeInfo(implicitArrayCreation).Type as IArrayTypeSymbol)?.ElementType`

Reply:
good catch!

Reply:
maybe more UTs are required then.

Reply:
Done

Reply:
Do the UTs cover differences in the array rank? I don't think so, and maybe the implementation as well:
```cs
void Method(params object[] args) { }

Method(new object[] { null }); // Noncompliant
// These cases might be FPs
Method(new object[,] { { null } }); // Compliant
Method(new[,] { { new object() } }); // Compliant

// This should work
Method(new object[][] { new[] { new object() } }); // Compliant
```

params array must be single-dimensional.

Reply:
We cover only jagged arrays.